docs/docker_host_ssh.md

```diff
@@ -0,0 +1,116 @@
+Reuse docker host SSH daemon
+============================
+
+It is possible to use the SSH daemon that runs on the docker host instead of using a separate SSH port for GitLab.
+
+## Setup
+
+Create the system `git` user if necessary (make sure it has a working shell):
+
+```bash
+useradd --system git --home-dir /srv/docker/gitlab/data
+```
+
+The home directory must point to GitLab data directory, as it takes `.ssh/authorized_keys` from there.
+
+Now create `gitlab-shell` home directory at the same location as in the container:
+
+```bash
+mkdir -p /home/git/gitlab-shell/bin
+```
+
+Create the shell proxy script at `/home/git/gitlab-shell/bin/gitlab-shell`:
+
+```bash
+#!/bin/bash
+# Proxy SSH requests to docker container
+sudo docker exec -i -u git gitlab sh -c "SSH_CONNECTION='$SSH_CONNECTION' SSH_ORIGINAL_COMMAND='$SSH_ORIGINAL_COMMAND' $0 $1"
+```
+
+Make it executable:
+
+```bash
+chmod +x /home/git/gitlab-shell/bin/gitlab-shell
+```
+
+Allow `git` user limited sudo access to execute a command inside GitLab docker container:
+
+```bash
+echo "git ALL=NOPASSWD: /usr/bin/docker exec -i -u git gitlab *" > /etc/sudoers.d/docker-gitlab
+```
+
+## Run GitLab
+
+Start the container, mapping the `uid` and `gid` of host `git` user:
+
+```bash
+docker run --name gitlab -it --rm [options] \
+    --env "USERMAP_UID=$(id -u git)" --env "USERMAP_GID=$(id -g git)" \
+    sameersbn/gitlab:8.11.3
+```
+
+## Access GitLab
+
+Use Docker host SSH server to access repositories:
+
+```bash
+git clone git@docker.host:group/project.git
+```
+
+## Advanced: Use default UID/GID in the container
+
+if you rely on the default `uid` and `git`, for example if you link GitLab to [sameersbn/redmine](https://github.com/sameersbn/docker-redmine) container, you can run the container without UID/GID mapping and use [incron](http://inotify.aiken.cz/?section=incron&page=about&lang=en) to keep `.ssh/authorized_keys` accessible by host `git` user:
```

Comment:
I think it's meant to be "if you rely on the default `uid` and `gid`" here? (note "gid" instead of "git").